enhancements/network/bgp-ovn-kubernetes.md

```diff
@@ -0,0 +1,701 @@
+* ECMP routing support within OVN for BGP learned routes.
+* Support for advertising user-defined networks via BGP as long as there is no subnet overlap over the default VRF.
+* Allowing for VRF-Lite type of VPN where the user maps interfaces on the host to user-defined VRFs/networks and
+advertises VPN routes via BGP sessions over said VRFs.
+
+### Non-Goals
+
+* Support of any other routing protocol.
+* Running separate BGPd instances per VRF network.
+* Providing any type of API or operator to automatically connect two Kubernetes clusters via L3VPN.
+* Replacing the support that MetalLB provides today for advertising service IPs.
+* Support for any other type of BGP speaker other than FRR.
+
+### Future Goals
+
+* Support EVPN configuration and integration with a user’s DC fabric, along with MAC-VRFs and IP-VRFs.
+* Support iBGP with route reflectors.
+* Potentially advertising other IP addresses, including the Kubernetes API VIP across the BGP fabric.
+
+## Proposal
+
+OVN-Kubernetes will leverage other projects that already exist to enable BGP in Linux. FRR will be used as the BGP
+speaker and already has EVPN support for native Linux constructs like Linux bridges, VRF devices, VXLAN tunnels, etc.
+FRR may need some code contributions to allow it to integrate with OVN and Open vSwitch. For FRR configuration, the
+MetalLB project has already started an API to be able to configure FRR: 
+[https://github.com/metallb/frr-k8s](https://github.com/metallb/frr-k8s). While some of the configuration support for
+FRR may be directly exposed by FRR-K8S API, it may also be the case that some intermediary CRD provided by
+OVN-Kubernetes is required to integrate OVN-Kubernetes networking concepts into FRR.
+
+Functionally, FRR will handle advertising and importing routes and configuring those inside a Linux VRF. OVN-Kubernetes
+will be responsible for listening on netlink and configuring OVN-Kubernetes logical routers with routes learned by FRR.
+
+### Workflow Description
+
+An admin has the ability to configure BGP peering and choose what networks to advertise. A tenant is able to define
+networks for their namespace, but requires admin permission in order to expose those networks onto the provider's BGP
+fabric. A typical workflow will be for a user or admin to create a user-defined network, and then the admin will be
+responsible to:
+
+1. If setting up VRF-Lite, do any host modifications necessary via NMState to enslave an IP interface to the
+   matching network VRF.
+2. Configure BGP peering via interacting with the FRR-K8S API for a given set of worker nodes. Also define filters for
+what routes should be received from the provider network.
+3. Create the OVN-Kubernetes RouteAdvertisements CR to configure what routes and where to advertise them.
+4. Verify peering and that routes have been propagated correctly.
+5. If desired, change the transport type from geneve on the network to none in order to no longer use an overlay.
```

Comment:
I know this enhancement is more telco-use-case-focused, but historically there have been lots of (non-telco) customers who wanted BGP _just_ so they could get rid of encapsulation, so having a really easy way to configure just that might be good...